### PR TITLE
Alias StackVector<int64_t, kMaxNdim> to Dims

### LINKED ISSUE #7343: Add type alias to StackVector<int64_t, kMaxNdim>
Relates to https://github.com/chainer/chainer/projects/2#card-16909303

Some other issue relates
- Argument of pooling kernels isn't typed `const&`
- Type of arguments named `stride` should be `Strides`
- Capacity of pad  should be double size of the dimension since it specifies begin and end pad
- There should be better name than Dims (some doesn't relate to dimensions)

### PR DESCRIPTION
**Pull request by take-cheeze**
This is a changes for: https://github.com/chainer/chainer/projects/2#card-16909303
closes #7343

There is few questions I want to ask:
- Is there any reason why `Dims` isn't typed `const&` in pooling kernels?
- Shouldn't the type of arguments named `stride` be `Strides` ?

### CODE REVIEW

**Comment by take-cheeze** [comment]
Other things that needs to be discussed:
- Shouldn't be used in pad since it needs double size of the dimension
- There should be better name than `Dims` (some doesn't relate to dimensions)

File: chainerx_cc/chainerx/dims.h [hunk 1]
```diff
@@ -8,10 +8,12 @@
 
 namespace chainerx {
 
-// Formatter to print StackVector containing integral elements as e.g. '[]' or '[1, 2, 3]'.
+typedef StackVector<int64_t, kMaxNdim> Dims;
```
Comment (kshitij12345) [inline chainerx_cc/chainerx/dims.h]: Wouldn't `using` be better ?

**Comment by hvy** [comment]
> Is there any reason why Dims isn't typed const& in pooling kernels?

It indeed seems better to use const&s. Could you send a separate PR, or file an issue?

> Shouldn't the type of arguments named stride be Strides ?

Do you mean the stride argument to the convolutions? These strides are different "strides" from what `Strides` is intended to represent. They are strides of the convolution kernel.

> Shouldn't be used in pad since it needs double size of the dimension

Hm, why is that? We currently don't allow specifying different padding values for e.g. left and right.

> There should be better name than Dims (some doesn't relate to dimensions)

Could you point me to where it doesn't relate to dimensions? The name "Dims" doesn't necessarily take the usage into account and is merely an alias for the underlying type (i.e. `StackVector<int64_t, kMaxNdim>`).

How about also renaming `CudaStackVector` to `CudaDims`?

**Comment by take-cheeze** [comment]
> Hm, why is that? We currently don't allow specifying different padding values for e.g. left and right.

Sorry it seems I confused with ONNX. Never mind.

**Comment by hvy** [comment]
Ah, it seems like ONNX allows fine grained control of these values. I did not know that.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
Hm, the Jenkins log was 852MB. Rerunning the build. 🤔

**Comment by hvy** [comment]
LGTM! The Jenkins failure seem unrelated.

**Pull request merged**